### PR TITLE
Remove dependency on Docker from MSBuild targets

### PR DESCRIPTION
**Pull request by natemcmaster**
Resolves https://github.com/aspnet/AspNetCore/issues/3653

Changes:
* This removes MSBuild targets which invoke `docker` commands to build
   deb and rpm installers
* Remove installer targets from the KoreBuild context. Put them into
   separate project files
* Simplify the targets used to build installers by reducing duplicate
   variable names and deeply nested MSBuild contexts
* Remove unused dependencies from the Docker build context

### CODE REVIEW

**Comment by natemcmaster** [comment]
Motivation for the changes:
* Preparing for arcade convergence (replacing of KoreBuild)
* Allows us to build rpm, deb, and other installers independently
* Source-build: make it easier to build only one type of Linux installer at a time
* Remove the dependency on Docker. As long as you have a Linux machine with the right requirements, you can build the repo.
* Simplify artifact flow on CI

cc @muratg

File: src/Installers/Rpm/Runtime.Rhel.rpmproj [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<!--
+  This project produces a RPM RHEL.7 installer for Linux x64 platforms only.
+-->
+<Project DefaultTargets="Build">
+  <Import Project="Rpm.props" />
+
+  <PropertyGroup>
+    <RpmPackageInstallRoot>/opt/rh/rh-dotnet20/root/usr/lib64/dotnet/</RpmPackageInstallRoot>
```
Comment (JunTaoLuo) [inline src/Installers/Rpm/Runtime.Rhel.rpmproj]: Hmm we need to check if this path is the correct one to use for 2.2. cc @leecow ?

Comment (natemcmaster) [reply src/Installers/Rpm/Runtime.Rhel.rpmproj]: This value is also what we use in 2.1. https://github.com/aspnet/AspNetCore/blob/release/2.1/build/SharedFxInstaller.targets#L122

**Review by JunTaoLuo — APPROVED** [review]

File: src/Installers/Rpm/Runtime.Rhel.rpmproj (hunk 1 above)
Comment (JunTaoLuo) [reply src/Installers/Rpm/Runtime.Rhel.rpmproj]: I asked about this in an email thread as well and I don't recall getting a definitive answer either.

**Review by muratg — APPROVED** [review]

**Pull request merged**